### PR TITLE
Allow empty array contract parameter

### PR DESCRIPTION
**Pull request by csmuller**

### CODE REVIEW

File: core/src/test/java/io/neow3j/contract/ContractParameterTest.java [hunk 1]
```diff
@@ -160,24 +161,18 @@ public void testArrayParamCreationFromObjects_failUnSupportedObject() {
...
 
     @Test
-    public void testArrayParam_Empty() {
-        exceptionRule.expect(IllegalArgumentException.class);
-        exceptionRule.expectMessage("At least one parameter is required");
-        array();
+    public void testArrayParam_Null() {
+        ContractParameter param = array((Object) null);
+        ContractParameter[] value = (ContractParameter[]) param.getValue();
+        assertThat(value.length, is(1));
```
Comment (gsmachado) [inline core/src/test/java/io/neow3j/contract/ContractParameterTest.java:174]: hmmm, I don't get it... why is this `1`?

A reference to a `null` Object is also in the array/stack?

Comment (csmuller) [reply core/src/test/java/io/neow3j/contract/ContractParameterTest.java:174]: I was also confused first. But `array((Object) null)` means that we'll actually have an array with one element and that element is of parameter type `ANY` with value `null`.

Comment (gsmachado) [reply core/src/test/java/io/neow3j/contract/ContractParameterTest.java:174]: hmmmmmm... not that intuitive, to be honest.

Can we document this somewhere? Maybe in a section called "types" (or similar) in neow3j.io website?

Comment (csmuller) [reply core/src/test/java/io/neow3j/contract/ContractParameterTest.java:174]: I think it's rather weird to do what i did in this test.
The intuitive way of getting an empty array would be `array()`, which works well. So i don't think there's a problem with the way the method works.
Anyway, I will extend the javadoc on `array(...)`.

Comment (gsmachado) [reply core/src/test/java/io/neow3j/contract/ContractParameterTest.java:174]: great!

**Review by gsmachado — APPROVED** [review]
💪  indeed this makes dev's life easier.

**Pull request merged**